### PR TITLE
Added fs-extra module

### PR DESCRIPTION
**Pull request by yomexzo**
We (pretty much @minasmart) 😄  realized we were adding a lot of `util` scripts like `recursive-rmdir`, `recursive-cpdir` and `recursive-mkdir` which are all available in `fs-extra` module (an extension of the native `fs` module which also exposes everything in the default `fs` module).

👀 @minasmart @tessalt @harismahmood89

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,6 +46,7 @@
     "eslint-config-nightmare-mode": "2.3.0",
     "estraverse-fb": "1.3.1",
     "fake-xml-http-request": "1.4.0",
+    "fs-extra": "^0.30.0",
```
Comment (minasmart) [inline package.json]: Please follow the convention and pin the exact version.

**Comment by tessalt** [comment]
🔥  nice!

File: scripts/build [hunk 2]
```diff
@@ -4,12 +4,12 @@
 
 require('./util/work-from-root')();
 
-var cleanDirectory = require('./util/clean-directory');
+var fse = require('fs-extra');
```
Comment (minasmart) [inline scripts/build]: use full words reflective of the package. no arbitrary contractions. `fsExtra`

File: package.json (hunk 1 above)
Comment (yomexzo) [inline package.json]: crap! my bad.

File: scripts/ci [hunk 3]
```diff
@@ -4,18 +4,17 @@
...
 
 var testBuildPath = '.dist-test';
 
-cleanDirectory(testBuildPath);
+fsExtra.removeSync(testBuildPath);
 
 buildProject(testBuildPath, function (buildStatus) {
   if (buildStatus === 0) {
-    var xunitWriteStream = fs.createWriteStream('tests.xml');
+    var xunitWriteStream = fsExtra.createWriteStream('tests.xml');
```
Comment (minasmart) [inline scripts/ci]: What benefit does the fsExtra `createWriteStream` have over the builtin one?

Comment (yomexzo) [inline scripts/ci]: `fsExtra`" is a drop in replacement for `fs`". It only adds new methods. So i figured no point requiring `fs` separately

Comment (minasmart) [inline scripts/ci]: I feel that that obfuscates where things come from. I'd prefer we use builtins wherever possible, and `fsExtra` for things only supplied by `fsExtra`

Comment (yomexzo) [inline scripts/ci]: 👍

Comment (yomexzo) [inline scripts/ci]: done

**Comment by minasmart** [comment]
👍

**Pull request merged**